bugbug/tools/code_review_generation.py

```diff
@@ -0,0 +1,754 @@
+                                            revision_id,
+                                            patch_id,
+                                            prompt_type,
+                                            diff_length_limit,
+                                            hunk_size,
+                                            comment_length,
+                                            generated_code_length,
+                                            metrics["precision"],
+                                            metrics["recall"],
+                                            metrics["f1"],
+                                            qualitative_feedback,
+                                            file_path,
+                                            comment.content,
+                                            comment.start_line,
+                                            comment.end_line,
+                                            relevant_diff,
+                                            generated_fix,
+                                            reference_fix,
+                                        ]
+                                    )
+
+                                counter += 1
+
+                    else:
+                        print(f"No relevant diff found for Comment ID {comment.id}.\n")
+
+
+def extract_revision_id_list_from_dataset(dataset_file):
+    revision_ids = []
+
+    with open(dataset_file, "r") as f:
+        for line in f:
+            data = json.loads(line)
+            revision_ids.append(data["revision_id"])
+
+    return revision_ids
+
+
+def calculate_metrics(reference_fix, generated_fix):
+    reference_tokens = reference_fix.split()
+    generated_tokens = generated_fix.split()
+
+    common_tokens = set(reference_tokens) & set(generated_tokens)
+    precision = len(common_tokens) / len(generated_tokens) if generated_tokens else 0
+    recall = len(common_tokens) / len(reference_tokens) if reference_tokens else 0
+    f1 = (2 * precision * recall) / (precision + recall) if (precision + recall) else 0
+
+    return {
+        "precision": precision,
+        "recall": recall,
+        "f1": f1,
+    }
+
+
+def find_fix_in_dataset(
+    revision_id,
+    initial_patch_id,
+    dataset_file,
+):
+    with open(dataset_file, "r") as f:
+        for line in f:
+            data = json.loads(line)
+            if (
+                data["revision_id"] == revision_id
+                and data["initial_patch_id"] == initial_patch_id
+            ):
+                return data["fix_patch_diff"]
+    return None
+
+
+def compare_fixes(revision_id, initial_patch_id, generated_fix, reference_fix):
+    if reference_fix:
+        metrics = calculate_metrics(reference_fix, generated_fix)
+        return metrics
+    else:
+        print(
+            f"No matching fix found in the dataset for Revision {revision_id} and Patch {initial_patch_id}."
+        )
+        return None
+
+
+def main():
```

Comment:
The CLI script should be separate and moved in the scripts folder (see for example scripts/code_review_tool_runner.py)

Reply:
Moved it to `scripts/code_review_generation_runner.py`: 12b0d7cc24ade49f718b10994a91ede5025b55e0

Will move the evaluation script execution to a separate script as well!

Reply:
Moved the evaluation script to `scripts/code_review_generation_evaluator_runner.py`: 1ddbbdc0ef62c6a4bcecbe7bbf39947f7d82ae75